packages/flutter/lib/src/cupertino/desktop_text_selection_toolbar_button.dart

```diff
@@ -37,26 +37,17 @@ class CupertinoDesktopTextSelectionToolbarButton extends StatefulWidget {
     super.key,
     required this.onPressed,
     required Widget this.child,
-  }) : buttonItem = null;
+  })  : buttonItem = null,
+        text = null;
 
   /// Create an instance of [CupertinoDesktopTextSelectionToolbarButton] whose child is
   /// a [Text] widget styled like the default Mac context menu button.
-  CupertinoDesktopTextSelectionToolbarButton.text({
+  const CupertinoDesktopTextSelectionToolbarButton.text({
     super.key,
-    required BuildContext context,
     required this.onPressed,
-    required String text,
-  }) : buttonItem = null,
-       child = Text(
```

Comment:
Can you explain why you needed to refactor this to create `child` in `build`?

Reply:
For the same reason it's done on [CupertinoTextSelectionToolbarButton](https://github.com/luccasclezar/flutter/blob/2f4ca584559b33581ba771bc530879d3d2d3d725/packages/flutter/lib/src/cupertino/text_selection_toolbar_button.dart#L119), the child needs to be rebuilt when certain things change. In this case, when hovered I check if it's better for the text to be black or white depending on the background color.